README.md

```diff
@@ -1,86 +1,149 @@
+following, we first list the dependencies, and then explain two ways of installing them.
+
+### List of dependencies
+The following is a list of dependencies required to build MIGraphX from source code. This list is also available in the
+requirement files ```dev-requirements.txt``` and ```requirements.txt```.
 
-## Prerequisites
 * [ROCm cmake modules](https://github.com/RadeonOpenCompute/rocm-cmake) **required**
 * [MIOpen](https://github.com/ROCmSoftwarePlatform/MIOpen) for running on the GPU
+* [rocBLAS](https://github.com/ROCmSoftwarePlatform/rocBLAS) for running on the GPU
 * [HIP](https://github.com/ROCm-Developer-Tools/HIP) for running on the GPU
 * [Protobuf](https://github.com/google/protobuf) for reading [onnx](https://github.com/onnx/onnx) files
 * [Half](http://half.sourceforge.net/) - IEEE 754-based half-precision floating point library
 * [pybind11](https://pybind11.readthedocs.io/en/stable/) - for python bindings
+* [ONNX 1.7.0](https://github.com/onnx/onnx) and [Pytest](https://github.com/pytest-dev/pytest) for running the ONN backend
+  tests 
 
-## Installing the dependencies
+Note: 1) We have to use ONNX version 1.7.0 since changes in ONNX 1.8.0 is incompatible with our current implementation. 
+ONNX 1.7.0 can be installed as ```pip3 install onnx==1.7.0```, and updates to support ONNX version 1.8.0 will come soon. 2)
+MIOpen and rocBLAS can be installed as ```sudo apt update && sudo apt install -y miopen-hip rocblas```.
 
-Dependencies can be installed using the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
+### Installing the dependencies
+There are two alternative ways to install the above dependencies:
 
-To install rbuild:
-```
-pip install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
-```
+#### Run a shell script [build_prereqs.sh](./tools/build_prereqs.sh).
+
+(Note: 1. You need the sudo to install the above dependencies. 2. All dependencies are installed in default locations in
+the system and are accessible by all users.)
+
+#### Use the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
 
-To build dependencies along with MIGraphX
+To install rbuild (sudo may be needed.):
 ```
-rbuild build -d depend --cxx=/opt/rocm/llvm/bin/clang++
+pip3 install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
 ```
-This builds dependencies in the subdirectory named depend and then builds MIGraphX using these dependencies.
+and pip3 can be installed as `sudo apt update && sudo apt install -y python3-pip`
 
-## Building MIGraphX from source
+Dependencies are listed in files `requirements.txt` and `dev-requirements.txt` in the project diretory. To build the
+dependencies,
+* In ROCm3.3:
 
-## Configuring with cmake
-
-First create a build directory:
+```
+rbuild develop -d depend -B build --cxx=/opt/rocm/bin/hcc
+```
 
+* In ROCm3.7 and later releases:
 
 ```
-mkdir build; 
-cd build;
+rbuild develop -d depend -B build --cxx=/opt/rocm/llvm/clang++
```

Comment:
This should be `rbuild build` or `rbuild package`. `rbuild develop` is only needed for developers.

Reply:
I just tried the command "rbuild build", that is the error message I got.
```
Successfully installed msgpack/msgpack-c
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is Clang 12.0.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/rocm/llvm/bin/clang++
-- Check for working CXX compiler: /opt/rocm/llvm/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:21 (find_package):
  By not providing "FindROCM.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "ROCM", but
  CMake did not find one.

  Could not find a package configuration file provided by "ROCM" with any of
  the following names:

    ROCMConfig.cmake
    rocm-config.cmake

  Add the installation prefix of "ROCM" to CMAKE_PREFIX_PATH or set
  "ROCM_DIR" to a directory containing one of the above files.  If "ROCM"
  provides a separate development package or SDK, be sure it has been
  installed.
```

Reply:
In my opinion, we should avoid this kind of errors. It is easy for an external user to give up

Reply:
The user needs to install rocm-cmake from the package manager.

Reply:
Rocm-cmake should be included in the `apt install` command.